packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (toolInstance.configuration.calculateStats) {
+      annotations[this._commonData.selectedIndex].invalidated = true;
+    }
+
+    triggerAnnotationModified(
+      annotations[this._commonData.selectedIndex],
+      element
+    );
+  };
+
+  /**
+   * Event handler for MOUSE_DRAG during the active loop.
+   *
+   * @param evt - The event
+   */
+  _dragCallback(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+    this._updateCursor(evt);
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length || !this.isActive) {
+      return;
+    }
+
+    const points =
+      annotations[this._commonData.selectedIndex].data.contour.polyline;
+
+    this._sculpt(eventData, points);
+  }
+
+  /**
+   * Attaches event listeners to the element such that is is visible, modifiable, and new data can be created.
+   * @param element - - The viewport element to attach event listeners to.
+   */
+  _activateSculpt(element: HTMLDivElement): void {
+    element.addEventListener(
+      Events.MOUSE_UP,
+      this._endCallback as EventListener
+    );
+    element.addEventListener(
+      Events.MOUSE_CLICK,
+      this._endCallback as EventListener
+    );
+    element.addEventListener(
+      Events.MOUSE_DRAG,
+      this._dragCallback as EventListener
+    );
+  }
+
+  /**
+   * Removes event listeners from the element.
+   * @param element - The viewport element to remove event listeners from.
+   */
+  _deactivateSculpt(element: HTMLDivElement): void {
+    element.removeEventListener(
+      Events.MOUSE_UP,
+      this._endCallback as EventListener
+    );
+    element.removeEventListener(
+      Events.MOUSE_CLICK,
+      this._endCallback as EventListener
+    );
+    element.removeEventListener(
+      Events.MOUSE_DRAG,
+      this._dragCallback as EventListener
+    );
+  }
+
+  _checkSpacing(
+    i: number,
+    points: Array<Types.Point3>,
+    indicesToInsertAfter: Array<number>,
+    maxSpacing: number
+  ): void {
+    const { element } = this._sculptData;
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const nextHandleIndex = this._getNextHandleIndex(i, points.length);
+
+    const currentCanvasPoint = viewport.worldToCanvas(points[i]);
+    const nextCanvasPoint = viewport.worldToCanvas(points[nextHandleIndex]);
+
+    const distanceToNextHandle = point.distanceToPoint(
+      currentCanvasPoint,
+      nextCanvasPoint
+    );
+
+    if (distanceToNextHandle > maxSpacing) {
+      indicesToInsertAfter.push(i);
+    }
+  }
+
+  _getNextHandleIndex(i: number, length: number): number {
+    if (i === length - 1) {
+      return 0;
+    }
+
+    return i + 1;
+  }
+
+  _getNextHandleIndexBeforeInsert(insertIndex: number, length: number): number {
```

Comment:
This one can just use the baseline contourIndex function, as it is identical to that - but it also works for negative values (which can be very handy).